ImperatorToCk3/Source/Imperator/ImperatorWorld.cpp

```diff
@@ -0,0 +1,108 @@
+#include "ImperatorWorld.h"
+#include "../Common/CommonFunctions.h"
+#include "../Common/Version.h"
+#include "../Configuration/Configuration.h"
+#include "Date.h"
+#include "Log.h"
+#include "OSCompatibilityLayer.h"
+#include "ParserHelpers.h"
+#include <ZipFile.h>
+#include <filesystem>
+#include <fstream>
+
+namespace fs = std::filesystem;
+
+ImperatorWorld::World::World(const Configuration& theConfiguration)
+{
+	LOG(LogLevel::Info) << "*** Hello Imperator, Roma Invicta! ***";
+	registerRegex("\\bSAV\\w*\\b", [](const std::string& unused, std::istream& theStream) {});
+	registerKeyword("version", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::singleString versionString(theStream);
+		ImperatorVersion = Version(versionString.getString());
+		Log(LogLevel::Info) << "<> Savegame version: " << versionString.getString();
+	});
+	registerKeyword("date", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::singleString dateString(theStream);
+		endDate = date(dateString.getString());
+		Log(LogLevel::Info) << "<> Date: " << dateString.getString();
+	});
+	registerKeyword("enabled_dlcs", [this](const std::string& unused, std::istream& theStream) {
```

Comment:
A word of warning. You really shouldn't need to read this. Doing custom procedures for all DLC specifics is a surefire way to go insane. Assume all DLCs are on for now. In future if something comes up that leaves a different footprint in the save, like Dharma changed government format in EU4, just map old values to new values.

Reply:
I disabled it for now.